core/src/oauth/credential.rs

```diff
@@ -0,0 +1,113 @@
+use core::fmt;
+use std::str::FromStr;
+
+use serde::{Deserialize, Serialize};
+
+use crate::oauth::store::OAuthStore;
+use crate::utils::{self, ParseError};
+use anyhow::Result;
+
+#[derive(Debug, Clone, Copy, Serialize, PartialEq, Deserialize)]
+#[serde(rename_all = "snake_case")]
+pub enum CredentialProvider {
+    Snowflake,
+}
+
+impl fmt::Display for CredentialProvider {
+    fn fmt(&self, f: &mut fmt::Formatter) -> fmt::Result {
+        write!(
+            f,
+            "{}",
+            serde_json::to_string(&self).unwrap().trim_matches('"')
+        )
+    }
+}
+
+impl FromStr for CredentialProvider {
+    type Err = ParseError;
+    fn from_str(s: &str) -> Result<Self, Self::Err> {
+        match serde_json::from_str(&format!("\"{}\"", s)) {
+            Ok(v) => Ok(v),
+            Err(_) => Err(ParseError::new()),
+        }
+    }
+}
+
+#[derive(Debug, Serialize, Clone)]
+pub struct Credential {
+    credential_id: String,
+    created: u64,
+    provider: CredentialProvider,
+    raw_json: serde_json::Value,
+}
+
+impl Credential {
+    pub fn new(
+        credential_id: String,
+        created: u64,
+        provider: CredentialProvider,
+        raw_json: serde_json::Value,
```

Comment:
credentials. Also in the current code this is a string not a JSON which is misleading here but should be OK once we fix the API

Reply:
Ok for naming.

On string vs json vs serialized json: 
If I understand correctly, we are converging toward expecting valid json (vs any string) but you also propose to use a real json format for the db column ?

However, don't we want to keep it encrypted ? meaning it has to stay a string, right ?

Reply:
I dont think we want to use a json column in db